File: WandleWheelhouse/src/WandleWheelhouse.Api/DTOs/Users/UserDetailDto.cs
using System;
using System.Collections.Generic; // Required for IList

namespace WandleWheelhouse.Api.DTOs.Users // Ensure namespace matches your structure
{
    #nullable enable // Enable nullable context for optional strings
    public class UserDetailDto
    {
        public string Id { get; set; } = string.Empty;
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string? PhoneNumber { get; set; }
        public bool EmailConfirmed { get; set; }
        public bool LockoutEnabled { get; set; }
        public DateTimeOffset? LockoutEnd { get; set; }

        // Address fields
        public string? AddressLine1 { get; set; }
        public string? AddressLine2 { get; set; }
        public string? City { get; set; }
        public string? PostCode { get; set; }
        public string? Country { get; set; }

        // Assigned Roles
        public IList<string> Roles { get; set; } = new List<string>();

        // --- ADD THIS PROPERTY ---
        public string? AvatarUrl { get; set; }
        // --- END ADD ---
    }
    #nullable disable // Disable nullable context if needed elsewhere
}

File: WandleWheelhouse/client/src/dto/Users/UserDetailDto.ts
// Location: src/dto/Users/UserDetailDto.ts

// Represents the detailed user information received from endpoints
// like /api/admin/users/{userId} or /api/users/me/profile
export interface UserDetailDto {
    id: string; // User ID (GUID as string)
    firstName: string;
    lastName: string;
    email: string;
    phoneNumber?: string | null;
    emailConfirmed: boolean;
    lockoutEnabled: boolean;
    lockoutEnd?: string | null; // DateTimeOffset often serialized as ISO string
  
    // Address fields
    addressLine1?: string | null;
    addressLine2?: string | null;
    city?: string | null;
    postCode?: string | null;
    country?: string | null;
  
    // Roles assigned to the user
    roles: string[];
  
    // Avatar URL (relative path from backend)
    avatarUrl?: string | null;
  }